overlays/xtensa_mtk_mt8195/gdb/gdb/xtensa-config.c

```diff
@@ -0,0 +1,504 @@
+  XTREG(235,975, 1, 1, 1,0x001b,0x0006,-2, 6,0x1010,b11,
+            0,0,&xtensa_mask11,0,0,0)
+  XTREG(236,976, 1, 1, 1,0x001c,0x0006,-2, 6,0x1010,b12,
+            0,0,&xtensa_mask12,0,0,0)
+  XTREG(237,977, 1, 1, 1,0x001d,0x0006,-2, 6,0x1010,b13,
+            0,0,&xtensa_mask13,0,0,0)
+  XTREG(238,978, 1, 1, 1,0x001e,0x0006,-2, 6,0x1010,b14,
+            0,0,&xtensa_mask14,0,0,0)
+  XTREG(239,979, 1, 1, 1,0x001f,0x0006,-2, 6,0x1010,b15,
+            0,0,&xtensa_mask15,0,0,0)
+  XTREG(240,980, 4, 4, 4,0x2005,0x0006,-2, 6,0x1010,psintlevel,
+            0,0,&xtensa_mask16,0,0,0)
+  XTREG(241,984, 1, 4, 4,0x2006,0x0006,-2, 6,0x1010,psum,
+            0,0,&xtensa_mask17,0,0,0)
+  XTREG(242,988, 1, 4, 4,0x2007,0x0006,-2, 6,0x1010,pswoe,
+            0,0,&xtensa_mask18,0,0,0)
+  XTREG(243,992, 2, 4, 4,0x2008,0x0006,-2, 6,0x1010,psring,
+            0,0,&xtensa_mask19,0,0,0)
+  XTREG(244,996, 1, 4, 4,0x2009,0x0006,-2, 6,0x1010,psexcm,
+            0,0,&xtensa_mask20,0,0,0)
+  XTREG(245,1000, 2, 4, 4,0x200a,0x0006,-2, 6,0x1010,pscallinc,
+            0,0,&xtensa_mask21,0,0,0)
+  XTREG(246,1004, 4, 4, 4,0x200b,0x0006,-2, 6,0x1010,psowb,
+            0,0,&xtensa_mask22,0,0,0)
+  XTREG(247,1008, 6, 4, 4,0x200c,0x0006,-2, 6,0x1010,mpunumentries,
+            0,0,&xtensa_mask23,0,0,0)
+  XTREG(248,1012, 4, 4, 4,0x2011,0x0006,-2, 6,0x1010,dbnum,
+            0,0,&xtensa_mask24,0,0,0)
+  XTREG(249,1016, 1, 4, 4,0x2014,0x0006, 1, 5,0x1010,ae_overflow,
+            0,0,&xtensa_mask25,0,0,0)
+  XTREG(250,1020,14, 4, 4,0x2015,0x0006, 1, 5,0x1010,ae_sar,
+            0,0,&xtensa_mask26,0,0,0)
+  XTREG(251,1024, 1, 4, 4,0x2016,0x0006, 1, 5,0x1010,ae_cwrap,
+            0,0,&xtensa_mask27,0,0,0)
+  XTREG(252,1028, 4, 4, 4,0x2017,0x0006, 1, 5,0x1010,ae_bitptr,
+            0,0,&xtensa_mask28,0,0,0)
+  XTREG(253,1032, 4, 4, 4,0x2018,0x0006, 1, 5,0x1010,ae_bitsused,
+            0,0,&xtensa_mask29,0,0,0)
+  XTREG(254,1036, 4, 4, 4,0x2019,0x0006, 1, 5,0x1010,ae_tablesize,
+            0,0,&xtensa_mask30,0,0,0)
+  XTREG(255,1040, 4, 4, 4,0x201a,0x0006, 1, 5,0x1010,ae_first_ts,
+            0,0,&xtensa_mask31,0,0,0)
+  XTREG(256,1044,27, 4, 4,0x201b,0x0006, 1, 5,0x1010,ae_nextoffset,
+            0,0,&xtensa_mask32,0,0,0)
+  XTREG(257,1048, 1, 4, 4,0x201c,0x0006, 1, 5,0x1010,ae_searchdone,
+            0,0,&xtensa_mask33,0,0,0)
+  XTREG(258,1052, 2, 4, 4,0x201d,0x0006, 1, 6,0x1010,roundmode,
+            0,0,&xtensa_mask34,0,0,0)
+  XTREG(259,1056, 1, 4, 4,0x201e,0x0006, 1, 6,0x1010,invalidflag,
+            0,0,&xtensa_mask35,0,0,0)
+  XTREG(260,1060, 1, 4, 4,0x201f,0x0006, 1, 6,0x1010,divzeroflag,
+            0,0,&xtensa_mask36,0,0,0)
+  XTREG(261,1064, 1, 4, 4,0x2020,0x0006, 1, 6,0x1010,overflowflag,
+            0,0,&xtensa_mask37,0,0,0)
+  XTREG(262,1068, 1, 4, 4,0x2021,0x0006, 1, 6,0x1010,underflowflag,
+            0,0,&xtensa_mask38,0,0,0)
+  XTREG_END
+};
+
+
+
+#ifdef XTENSA_CONFIG_INSTANTIATE
+XTENSA_CONFIG_INSTANTIATE(rmap,16)
+#endif
+
```

Comment:
Need to add:
```
xtensa_gdbarch_tdep xtensa_tdep (rmap);
```
or else it won't build.

Reply:
Sorry, I'm lost on that requirement.  Where exactly do I put that and why?  :)

This is just the upstream file from the Cadence SDK (with some path munging via that make-overlay.sh script at foss-xtensa).  Is this something they've forgotten or that we need to maintain and didn't document?  Or am I looking at the wrong place?

Reply:
Ah, OK.  Skew vs. upstream gdb APIs.  Found the other commits in the tree.

We really need to write a "overlay importer" layer or something.  To add to the list with "make dynamic configs work" I guess.  Managing this is sort of a mess for someone doing it the first time, and we're juggling a ton of these toolchains these days.